### PR TITLE
feat: 🎸 Add condensed style for dropdown

### PR DESCRIPTION
**Pull request by calcaide**
First iteration to add a condensed style for dropdown.

✅ Closes: ICU-1431

## Screenshots:

Condensed dropdown close:
Screen Shot 2021-06-16 at 10 47 43 AM

Condensed dropdown open:
Screen Shot 2021-06-16 at 10 48 37 AM

## Comments/Questions:
- The dropdown button `<details>` has a background (light grey) when the dropdown is open. I am pretty sure we do not want that, but just to confirm. Should I remove that background? It is visible on the condensed dropdown open screenshot.
- I did not use `<dropdown.separator />` as a separator between `sections`. I just add a `border-bottom` at section level. As per figma design I understand this is a better approach for this use case, but happy to change approach and use the `<dropdown.separator />`. Let me know your thoughts on this.
- Within the dropdown content, I applied lateral padding per `<section>` rather than the div `.rose-dropdown-content`. Why? because the border bottom of the section has to cover the whole `.rose-dropdown-content` width.
- Once the dropdown is open, if you click outside (anywhere) the dropdown closes. Do we want that behaviour?
- Once the dropdown is open, if you click within the dropdown it closes. Do we want that behaviour?
- Be aware on the ID field (first field) I did not add `copyable` component yet because this is done in Rose and `copyable` is an ad hoc component. Once we merge this PR I can test to add the component to see how it looks.
- Be aware I DID NOT change the storybook page reflecting these changes yet. I will do that once we are ok with these changes.

### CODE REVIEW

File: addons/rose/addon/styles/rose/components/dropdown/_item.scss [hunk 1]
```diff
@@ -32,11 +32,6 @@ $checkmark: url("data:image/svg+xml;utf-8,<svg viewBox='0 0 24 24' fill='%231563
       padding-right: sizing.rems(xl);
       font-weight: normal;
 
-      // &::before,
```
Comment (calcaide) [inline addons/rose/addon/styles/rose/components/dropdown/_item.scss:35]: This was old commented code. I assume we do not need it anymore.

File: addons/rose/addon/styles/rose/components/dropdown/_item.scss [hunk 2]
```diff
@@ -57,11 +52,6 @@ $checkmark: url("data:image/svg+xml;utf-8,<svg viewBox='0 0 24 24' fill='%231563
         ~ .rose-form-radio-label {
           color: var(--action);
           background-color: var(--action-subtler-1h);
-
-          // background-image: #{$checkmark};
```
Comment (calcaide) [inline addons/rose/addon/styles/rose/components/dropdown/_item.scss:61]: This was old commented code. I assume we do not need it anymore.

File: addons/rose/addon/styles/rose/components/form/_fieldset.scss [hunk 3]
```diff
@@ -3,7 +3,7 @@
 
 .rose-form-fieldset {
   margin-bottom: sizing.rems(m);
-  
+
```
Comment (calcaide) [inline addons/rose/addon/styles/rose/components/form/_fieldset.scss:6]: Formatting change (after running `yarn run format`

File: addons/rose/tests/dummy/app/templates/index.hbs [hunk 4]
```diff
@@ -1,196 +1,50 @@
-<h2>Condensed Table</h2>
```
Comment (calcaide) [inline addons/rose/tests/dummy/app/templates/index.hbs:1]: Ignore all the changes in this file, this is just the dummy test for development purposes.

**Review by randallmorey — APPROVED** [review]

**Pull request merged**